support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -417,9 +425,16 @@ AccessControlled accessControlled() {
                         cookie = UUID.randomUUID().toString();
                         // Switches the label to a self-label, so if the executable is killed and restarted via ExecutorPickle, it will run on the same node:
                         label = computer.getName();
+
                         EnvVars env = computer.getEnvironment();
                         env.overrideAll(computer.buildEnvironment(listener));
                         env.put(COOKIE_VAR, cookie);
+                        if (exec.getOwner() instanceof MasterComputer) {
+                            env.put("NODE_NAME", "master");
```

Comment:
When you are forced to copy rather than reuse code, it is a good idea to include a `TODO` comment noting the origin (in this case, [`CoreEnvironmentContributor`](https://github.com/jenkinsci/jenkins/blob/9c443c8d5bafd63fce574f6d0cf400cd8fe1f124/core/src/main/java/jenkins/model/CoreEnvironmentContributor.java#L60)), in case a future refactoring makes reuse possible.